ikarus/assembler/assemblerwrappers.hh

```diff
@@ -0,0 +1,228 @@
+// SPDX-FileCopyrightText: 2021-2024 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+/**
+ * \file assemblerwrappers.hh
+ * \brief Defines the base classes for scalar, vector, and matrix wrappers to assemblers
+ */
+
+#pragma once
+
+#include <utility>
+
+#include <dune/functions/backends/istlvectorbackend.hh>
+
+#include <ikarus/assembler/dirichletbcenforcement.hh>
+#include <ikarus/assembler/interface.hh>
+#include <ikarus/finiteelements/fehelper.hh>
+#include <ikarus/finiteelements/ferequirements.hh>
+#include <ikarus/utils/concepts.hh>
+#include <ikarus/utils/dirichletvalues.hh>
+
+namespace Ikarus {
+/**
+ * \brief Base class for a wrapper to a scalar assembler.
+ * \tparam Wrapper Type of the wrapper to an assembler
+ * \tparam Assembler Type of the assembler
+ */
+template <template <typename> typename Wrapper, typename Assembler>
+struct ScalarWrapperBase
```

Comment:
These are not wrapper basis but actually the basis for the manipulator, which is one implementation of a wrapper. Therfeore these classes should be named differently.
and also the file

Reply:
What about just having `ScalarWrapper, VectorWrapper` and `MatrixWrapper` in a file called `wrapperinterface.hh`? And rename the other `interface.hh` to `assemblerinterface.hh`? Or do you have something else in mind?

Reply:
The contain the implementation of the manipulator. Therefore, they are no interface classes. Therefore, they are just base classes for the implementation. But this I changed anyway.